applications/crossbar/src/modules/cb_access_lists.erl

```diff
@@ -0,0 +1,186 @@
+%% @end
+%%--------------------------------------------------------------------
+-spec validate(cb_context:context()) -> cb_context:context().
+validate(Context) ->
+    validate_acls(Context, cb_context:req_verb(Context)).
+
+-spec validate_acls(cb_context:context(), http_method()) -> cb_context:context().
+validate_acls(Context, ?HTTP_GET) ->
+    validate_get_acls(Context, thing_doc(Context));
+validate_acls(Context, ?HTTP_POST) ->
+    cb_context:validate_request_data(<<"access_lists">>, Context, fun validate_set_acls/1);
+validate_acls(Context, ?HTTP_DELETE) ->
+    validate_delete_acls(Context, thing_doc(Context)).
+
+-spec thing_doc(cb_context:context()) -> api_object().
+-spec thing_doc(cb_context:context(), ne_binary()) -> api_object().
+thing_doc(Context) ->
+    case cb_context:req_nouns(Context) of
+        [{<<"access_lists">>, []}, {<<"accounts">>, [AccountId]} | _] ->
+            lager:debug("loading access lists from account: '~s'", [AccountId]),
+            thing_doc(Context, AccountId);
+        [{<<"access_lists">>, []}, {<<"devices">>, [DeviceId]} | _] ->
+            lager:debug("loading access lists from device: '~s'", [DeviceId]),
+            thing_doc(Context, DeviceId);
+        _Nouns -> 'undefined'
+    end.
+
+thing_doc(Context, ThingId) ->
+    Context1 = crossbar_doc:load(ThingId, Context),
+    case cb_context:resp_status(Context1) of
+        'success' -> cb_context:doc(Context1);
+        _Status ->
+            lager:debug("failed to load thing ~s", [ThingId]),
+            'undefined'
+    end.
+
+-spec validate_get_acls(cb_context:context(), api_object()) -> cb_context:context().
+validate_get_acls(Context, 'undefined') ->
+    crossbar_util:response_faulty_request(Context);
+validate_get_acls(Context, Doc) ->
+    AccessLists = wh_json:get_value(<<"access_lists">>, Doc, wh_json:new()),
+    crossbar_util:response(AccessLists, Context).
+
+-spec validate_delete_acls(cb_context:context(), api_object()) -> cb_context:context().
+validate_delete_acls(Context, 'undefined') ->
+    crossbar_util:response_faulty_request(Context);
+validate_delete_acls(Context, Doc) ->
+    crossbar_util:response(wh_json:new()
+                           ,cb_context:set_doc(Context
+                                               ,wh_json:delete_key(<<"access_lists">>, Doc)
+                                              )).
+
+-spec validate_set_acls(cb_context:context()) ->
+                                    cb_context:context().
+-spec validate_set_acls(cb_context:context(), wh_json:object(), api_object()) ->
+                                    cb_context:context().
+validate_set_acls(Context) ->
+    lager:debug("access lists data is valid, setting on thing"),
+    validate_set_acls(Context, cb_context:doc(Context), thing_doc(Context)).
+
+validate_set_acls(Context, _, 'undefined') ->
+    lager:debug("no doc found"),
+    crossbar_util:response_faulty_request(Context);
```

Comment:
Should probably return a 404 since Thing is not found. See `cb_context:add_system_error('bad_identifier', JObj, Context)` in places like `cb_devices_v2`.